src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -683,6 +685,23 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
             _incomingBuffer.Discard(frameHeader.Length);
         }
 
+        internal async Task FlushAsync() // no cancellation token: we don't want individual operation tokens applying to the whole connection
+        {
+            if (NetEventSource.IsEnabled) Trace("");
+            await AcquireWriteLockAsync(default).ConfigureAwait(false);
+            try
+            {
+                if (_outgoingBuffer.ActiveSpan.Length > 0)
+                {
+                    await FlushOutgoingBytesAsync().ConfigureAwait(false);
```

Comment:
We don't actually need to await this. We could do what FinishWrite does, which is to simply store the Task in _inProgressWrite. 

In fact, we might want to move that logic to FlushOutgoingBytesAsync and then make it non-async.

Reply:
> In fact, we might want to move that logic to FlushOutgoingBytesAsync and then make it non-async.

Actually, we await FlushOutgoingBytesAsync in StartWriteAsync, so we can't do this. But we could still do the _inProgressWrite thing.